### PR TITLE
Interactive title and description

### PR DESCRIPTION
**Pull request by brjadams**
Inject settings into APISchema, add title and description from settings object into Interactive API templates.

### CODE REVIEW

**Comment by brjadams** [comment]
Need to add default settings/schema to test.

File: apistar/apischema.py [hunk 1]
```diff
@@ -14,16 +14,20 @@
...
     Route, RoutesConfig, primitive_types, schema_types, walk
 )
+from apistar.settings import Settings
 from apistar.templating import Templates
 
 
 class APISchema(Document):
     @classmethod
-    def build(cls, app: App, base_url: http.URL=None):
+    def build(cls, app: App, settings: Settings=Settings({}), base_url: http.URL=None):
```
Comment (tomchristie) [inline apistar/apischema.py]: In preference to this I'd suggest default to `None` instead, and do a...

    if settings is None:
        settings = Settings({})

File: apistar/templates/docs/layout/content.html [hunk 2]
```diff
@@ -1,9 +1,7 @@
 <div class="row intro">
 <div class="col-md-6 intro-title">
     <h1>{{ document.title|default('API Star', True) }}</h1>
-    {% if document.description %}
-    <p>{{ document.description }}</p>
-    {% endif %}
+    <p>{{ document.description|default('description', True) }}</p>
```
Comment (tomchristie) [inline apistar/templates/docs/layout/content.html]: Probably wrap `{% if document.description %}` around this `<p>` block - we don't want to provide a default text string here.

File: apistar/apischema.py [hunk 3]
```diff
@@ -14,16 +14,20 @@
...
         routes = app.routes
         url = get_schema_url(routes, base_url)
         content = get_schema_content(routes)
-        return cls(url=url, content=content)
+        schema = settings.get("SCHEMA", {})
+        title = schema.get("TITLE", None)
+        description = schema.get("DESCRIPTION", None)
```
Comment (tomchristie) [inline apistar/apischema.py]: Probably prefer single quotes where possible (no particular reason, just prefer that stylistically)

Comment (brjadams) [inline apistar/apischema.py]: understood.

File: apistar/templates/docs/layout/content.html (hunk 2 above)
Comment (brjadams) [inline apistar/templates/docs/layout/content.html]: I believe it was this way previously. Will clean this up

**Comment by ssteinerx** [comment]
Does this also address/close [Integrating `schema.description` with docs?](#211)?

**Comment by tomchristie** [comment]
@ssteinerx I'm not sure that you meant to link back to this same ticket?

**Comment by ssteinerx** [comment]
Whoops, I was trying to link from the pull request about it and got the wrong tab!

**Comment by tomchristie** [comment]
That ticket is regarding field descriptions, rather than the top-level overall description.
Related, but not the same, no. :)

**Comment by ssteinerx** [comment]
@tomchristie Ah, thanks, sorry about that, just re-read the code in this pull request.  Just started using this yesterday and have only poked around enough to realize that my descriptions weren't coming out in the /docs.

**Comment by tomchristie** [comment]
I think this is a valid approach yup. If it gets refactored for 0.2, and gets appropriate docs, then I don't see why it shouldn't go in. Thanks!

**Comment by tomchristie** [comment]
Closed via https://github.com/encode/apistar/commit/4c95797a2c13d0b379c8f4027c0686d8f4442532

Does still need documenting tho.

**Pull request closed without merging**